lib/common/sanitizer.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = SanitizerFactory;
+
+SanitizerFactory.$provide = 'Sanitizer';
+SanitizerFactory.$inject = [
+    'Services.Encryption'
+];
+
+function SanitizerFactory (encryption) {
+    /**
+     * encrypt sensitve information.
+     * @constructor
+     */
+    //function Sanitizer () {
+    //}
+
+    /**
+     *
+     */
+    return function Sanitizer (workflow) {
+        //workflow = JSON.parse(workflow);
+        var sanitizedworkflow = workflow;
+        var walked = [];
+        var stack = [ { workflow : workflow , stack : '' } ];
+        while ( stack.length > 0 ) {
+            var item = stack.pop();
+            var workflow  = item.workflow;
+            for ( var property in workflow ) {
+                if ( workflow.hasOwnProperty( property )) {
+                    if ( typeof workflow[ property ] === "object" ) {
+                        var alreadyFound = false;
+                        for ( var i = 0; i < walked.length; i++ ) {
+                            if ( walked[i] === workflow[ property ] ) {
+                                alreadyFound = true;
+                                break;
+                            }
+                        }
+                        if ( !alreadyFound ) {
+                            walked.push( workflow[ property ]);
+                            stack.push( { workflow : workflow[ property ], stack: item.stack + '.' + property });
+                        }
+                    } else {
+                        if ( property.match( /password/i ) || property.match(/community/i)) {
```

Comment:
There is a constant array of redaction patterns you can use for this. See https://github.com/RackHD/on-core/blob/master/lib/models/obms.js#L36